### ISSUE TITLE
Removing Chloroplast and Mitochondria after assigning taxonomy

### ISSUE THREAD

**Issue by sandipansamaddar** [benjjneb/dada2#892]
Hi,
Sorry for opening an issue again.

I proceeded with the files for taxonomic classification and it worked well. I am stuck in one command where I am trying to remove both Chloroplast and Mitochondria from the files. I wish to remove them from both the count and the tax file. I have the seqtab.nochim and taxa file. As I am not an expert so can you please help me on that?

`seqtab.nochim <- removeBimeraDenovo(seqtab, verbose=T, multithread= TRUE)`
`taxa <- addSpecies(taxa,"silva_species_assignment_v132.fa.gz", tryRC = T)`

Thank you,

Sandipan

**Comment by benjjneb** [comment]
Did you find chloroplasts/mitochondria? You'll just want to identify those taxa, and then remove them from the tables.

```
st <- seqtab.nochim[,!is.mito]
tt <- taxa[!is.mito,]
```

**Comment by sandipansamaddar** [comment]
Thanks for the reply.

Yes I found both Chloroplasts and Mitochondria in the taxonomy file. So what I did was:

`is.chloro <- taxa[,"Order"] %in% "Chloroplast"
seqtab.nochloro <- seqtab.nochim[,!is.chloro]
dim(seqtab.nochloro)`
This removed the Chloroplast sequences from the count file. But when I put the same for the taxa file:
`taxa.nochloro <- taxa[,!is.chloro]`

it shows me error "((subscript) logical subscript too long)"

I encounter similar errors while trying to remove Mitochondria after removing Chloroplast.

I am not sure why it pops up.

Thanks,

Sandipan
